extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java

```diff
@@ -178,82 +186,26 @@ void nativeImageConfiguration(
         resource.produce(
                 new NativeImageResourceBuildItem(getChangeLogs(dataSourceNames, liquibaseBuildConfig).toArray(new String[0])));
 
-        Stream.of(liquibase.change.Change.class,
-                liquibase.changelog.ChangeLogHistoryService.class,
-                liquibase.changeset.ChangeSetService.class,
-                liquibase.database.Database.class,
-                liquibase.database.DatabaseConnection.class,
-                liquibase.datatype.LiquibaseDataType.class,
-                liquibase.diff.compare.DatabaseObjectComparator.class,
-                liquibase.diff.DiffGenerator.class,
-                liquibase.diff.output.changelog.ChangeGenerator.class,
-                liquibase.executor.Executor.class,
-                liquibase.license.LicenseService.class,
-                liquibase.lockservice.LockService.class,
-                liquibase.logging.LogService.class,
-                liquibase.parser.ChangeLogParser.class,
-                liquibase.parser.LiquibaseSqlParser.class,
-                liquibase.parser.NamespaceDetails.class,
-                liquibase.parser.SnapshotParser.class,
-                liquibase.precondition.Precondition.class,
-                liquibase.report.ShowSummaryGenerator.class,
-                liquibase.serializer.ChangeLogSerializer.class,
-                liquibase.serializer.SnapshotSerializer.class,
-                liquibase.servicelocator.ServiceLocator.class,
-                liquibase.snapshot.SnapshotGenerator.class,
-                liquibase.sqlgenerator.SqlGenerator.class,
-                liquibase.structure.DatabaseObject.class,
-                liquibase.logging.mdc.MdcManager.class)
-                .forEach(t -> consumeService(t, (serviceClass, implementations) -> {
-                    services.produce(
-                            new ServiceProviderBuildItem(serviceClass.getName(), implementations.toArray(new String[0])));
-                    reflective.produce(ReflectiveClassBuildItem.builder(implementations.toArray(new String[0]))
-                            .constructors().methods().build());
-                }));
-
         // Register Precondition services, and the implementation class for reflection while also registering fields for reflection
-        consumeService(liquibase.precondition.Precondition.class, (serviceClass, implementations) -> {
-            services.produce(new ServiceProviderBuildItem(serviceClass.getName(), implementations.toArray(new String[0])));
+        consumeService("liquibase.precondition.Precondition", (serviceClass, implementations) -> {
```

Comment:
I suggested to use `liquibase.precondition.Precondition.class.getName()` (and same for the others, for instance in `RUNTIME_INITIALIZATION_SERVICES`). Keep in mind that we don't manage the Liquibase development so better get a compilation warning if they change something in this area.

It's a bit more nuanced than that because we sometimes use the strings because either the classes are not accessible or because we want to avoid loading them if we can, but in this very case, I think it's better to use `.class.getName()`.

Reply:
I understand your reasoning here, but doesn't this contradict, what you wrote above?

> I think I would avoid transforming this to a class then a String again. You can make `runtimeInitializationServices` a set of strings and use `liquibase.configuration.ConfigurationValueProvider.class`.

I'm fine with restoring this to the class-based version, but I'd like to avoid any misunderstanding.

Reply:
OK, i see. I will change the code accordingly.